app/models/poll.rb

```diff
@@ -16,4 +16,11 @@ def taker
     self.user.username
   end
 
+  def survey_name
+    self.survey.name
+  end
+
+  def questions
+    self.survey.questions
+  end
```

Comment:
Lines 23-25 should not be necessary.  You should be able to retrieve the poll's questions at poll.questions automatically, since we have a designed a "has_many" association for questions through survey on the Poll model.